commcare_connect/form_receiver/processor.py

```diff
@@ -185,24 +183,42 @@ def process_deliver_unit(user, xform: XForm, app: CommCareApp, opportunity: Oppo
     ):
         user_visit.status = VisitValidationStatus.over_limit
     flags = []
+    opportunity_flags = opportunity.opportunityverificationflags
     if counts["entity"] > 0:
         user_visit.status = VisitValidationStatus.duplicate
-        flags.append(["duplicate", "A beneficiary with the same identifier already exists"])
-    if xform.metadata.duration < datetime.timedelta(seconds=60):
+        if opportunity_flags.duplicate:
+            flags.append(["duplicate", "A beneficiary with the same identifier already exists"])
+    if opportunity_flags.duration > 0 and xform.metadata.duration < datetime.timedelta(
+        minutes=opportunity_flags.duration
+    ):
         flags.append(["duration", "The form was completed too quickly."])
     if xform.metadata.location is None:
-        flags.append(["gps", "GPS data is missing"])
+        if opportunity_flags.gps:
+            flags.append(["gps", "GPS data is missing"])
     else:
-        user_visits = UserVisit.objects.filter(opportunity=opportunity, deliver_unit=deliver_unit).values("location")
-        cur_lat, cur_lon, *_ = xform.metadata.location.split(" ")
-        for visit in user_visits:
-            if visit.get("location") is None:
-                continue
-            lat, lon, *_ = visit["location"].split(" ")
-            dist = distance((lat, lon), (cur_lat, cur_lon))
-            if dist.m <= 10:
-                flags.append(["location", "Visit location is too close to another visit"])
-                break
+        if opportunity_flags.location > 0:
+            user_visits = UserVisit.objects.filter(opportunity=opportunity, deliver_unit=deliver_unit).values(
+                "location"
+            )
+            cur_lat, cur_lon, *_ = xform.metadata.location.split(" ")
+            for visit in user_visits:
+                if visit.get("location") is None:
+                    continue
+                lat, lon, *_ = visit["location"].split(" ")
+                dist = distance((lat, lon), (cur_lat, cur_lon))
+                if dist.m <= 10:
+                    flags.append(["location", "Visit location is too close to another visit"])
+                    break
+    if (
+        opportunity_flags.form_submission_start
+        and opportunity_flags.form_submission_start.time() < xform.metadata.timeStart.time()
```

Comment:
are these checking server timezone or phone timezone?

Reply:
Thanks for pointing this out. Fixed here. https://github.com/dimagi/commcare-connect/pull/301/commits/fef6d6d915c80072d3ea01fc1bcd150cb9fa1501

Reply:
Also on a further note, we are using the UTC timezone right now. It would make sense to add a help text under the form submission section to indicate that the start and end times need to be adjusted according to the UTC time.

Reply:
Oops, I think you may not have been clear, and you took this the opposite way I mean. Because there can be a large delay between when a form is filled out and received by connect (for example if they need to wait for better data coverage to submit, or repeaters are slow), we need to use the time the form was filled not received to check whether it is flagged. I just wanted to make sure that the we were careful to make sure the entered time and form time were in the same timezone (not sure if what we get from the phone is in UTC or local time).

Reply:
The phone timezone is UTC.

Reply:
Fixed. https://github.com/dimagi/commcare-connect/pull/301/commits/66688cdd2e7e6621e0dae74f763ae7be821fa180